app/src/main/res/layout/activity_ipcam_snapshot.xml

```diff
@@ -19,13 +23,20 @@
             android:layout_alignParentBottom="true"
             android:layout_alignParentEnd="true"
             android:layout_alignParentRight="true"
+            android:visibility="gone"
             stream:srcCompat="@mipmap/ic_launcher" />
 
-        <com.github.niqdev.mjpeg.MjpegSurfaceView
-            android:id="@+id/mjpegViewSnapshot"
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            stream:type="stream_default" />
+        <TextView
+            android:id="@+id/record_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:visibility="gone"
+             android:layout_centerHorizontal="true"
+            android:layout_marginTop="20dp"
+            android:textSize="18sp"
+            android:textColor="@color/red"
```

Comment:
Please use `android:textColor="@android:color/holo_red_dark"` and remove your `red` from `colors.xml` to keep sample as simple as possible

Reply:
> @wxkly8888 this is great, thanks for your contribution!
> 
> You added all the main recording logic in the sample activity `IpCamSnapshotActivity`, this is meant only to show the usage of the libray, the core functionality should be all in the lib. Like `saveBitmapToFile`, `startRecording` and `stopRecording` should be all exposed somehow from the lib. Otherwise, this functionality won't be useful to other users. wdyt? Also, I would document it in the readme.
> 
> @hannesa2 do you have any suggestions on this?

here I want to share my point, if we add these methods to the library.  then we need to meet the needs of different users, same user may want to save image as png or mp4.  and same users may want to save files in Gallery or other app shared directory (need to implement in different ways).
In this PR, I expose a method from the lib **frameCapturedWithByteData(byte[] imageByte,byte[] header)**, so you can get the raw data . then you can assemble it into any format files you want. and in IpCamSnapshotActivity you can get example how to use it.

Reply:
> same user may want to save image as png or mp4. and same users may want to save files in Gallery or other app shared directory (need to implement in different ways)

What about a `MjpegUtils` then? If you split the methods and allow the user to compose them, it would be way more useful. Your example is great, don't get me wrong, I only think that it should be more flexible and reusable

Reply:
yeah, a good ieda. I have extracted methods from IpCamSnapshotActivity into MjpegRcordingHandler class. and put it into the library. please review it.
and currently only support MJPEG and jpg format.  saving as MP4 is not easy work. also saving files into Gallery need additional permissions logic. maybe I can add it later if i have time.  I am just making a start here :smile: